src/compiler/crystal/codegen/codegen.cr

```diff
@@ -361,6 +362,13 @@ module Crystal
         mod = info.mod
         push_debug_info_metadata(mod) unless @debug.none?
 
+        if ENV["CRYSTAL_DEBUG_LL_DUMP"]?
+          puts "Module #{name}:"
+          mod.functions.each do |func|
+            puts func.body_to_ll_string
+          end
+        end
+
```

Comment:
Can this be removed? Or what's the diffrence between that and the other `DUMP` env var we have?

Reply:
This is my extended version of IR. It expands Debug info next to the IR line:

```
def __crystal_main(i32 %argc, i8** %argv):
Block alloca:
          %a = alloca %"Array(Int16 | Int32 | Int64 | Int8 | UInt16 | UInt32 | UInt64 | UInt8 | Nil)"*, !dbg !538 -> metadata: !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>) = !DILocation(line:
          call void @llvm.dbg.declare(metadata %"Array(Int16 | Int32 | Int64 | Int8 | UInt16 | UInt32 | UInt64 | UInt8 | Nil)"** %a, metadata !465, metadata !DIExpression()), !dbg !539 -> metadata: !DILoc
          %b = alloca [256 x i8], !dbg !538 -> metadata: !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>) = !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>)
          call void @llvm.dbg.declare(metadata [256 x i8]* %b, metadata !493, metadata !DIExpression()), !dbg !540 -> metadata: !DILocation(line: 13, column: 1, scope: <0x7fa98c608920>) = !DILocation(line
          %c = alloca %"Tuple(Int8, UInt8)", !dbg !538 -> metadata: !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>) = !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>)
          call void @llvm.dbg.declare(metadata %"Tuple(Int8, UInt8)"* %c, metadata !497, metadata !DIExpression()), !dbg !541 -> metadata: !DILocation(line: 14, column: 1, scope: <0x7fa98c608920>) = !DILo
          %d = alloca %"NamedTuple(str: String, val: UInt8)", !dbg !538 -> metadata: !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>) = !DILocation(line: 1, column: 1, scope: <0x7fa98c608920>)
          call void @llvm.dbg.declare(metadata %"NamedTuple(str: String, val: UInt8)"* %d, metadata !502, metadata !DIExpression()), !dbg !542 -> metadata: !DILocation(line: 15, column: 1, scope: <0x7fa98
          %0 = alloca %"->" -> metadata: N/A
...
```
It simplifies IR file analysis.

Reply:
I removed it. I will make full blown IR viewer for that as separate project.